pkg/process/checks/process_discovery_check.go

```diff
@@ -104,7 +104,20 @@ func (d *ProcessDiscoveryCheck) Run(nextGroupID func() int32, _ *RunOptions) (Ru
 		NumCpus:     calculateNumCores(d.info.SystemInfo),
 		TotalMemory: d.info.SystemInfo.TotalMemory,
 	}
-	procDiscoveryChunks := chunkProcessDiscoveries(pidMapToProcDiscoveries(procs, d.userProbe, d.scrubber), d.maxBatchSize)
+
+	procDiscoveries := pidMapToProcDiscoveries(procs, d.userProbe, d.scrubber)
+
+	// For no chunking, set max batch size as number of process discoveries to ensure one chunk
+	if options != nil && options.NoChunking {
+		oldMaxBatchSize := d.maxBatchSize
+		d.maxBatchSize = len(procDiscoveries)
+
+		defer func() {
+			d.maxBatchSize = oldMaxBatchSize
+		}()
+	}
```

Comment:
Instead of using defer funcs, what do you think about:
```go
runMaxBatchsize := d.maxBatchSize
if options != nil && options.NoChunking {
		runMaxBatchsize := d.maxBatchSize
}
procDiscoveryChunks := chunkProcessDiscoveries(procDiscoveries, runMaxBatchSize)
```

Reply:
You could do similar for the process check

Reply:
For process discovery check I totally agree this is more clean and in line with how container checks sets the batch size - I will change this in the next commit. 

For process check, I will still use the defer func since the d.maxBatchSize variable is not passed as argument for chunking, but directly accessed as a member field within nested function calls of the Run() function (if I create this new runBatchMaxSize variable, it would have to be passed down many layers and bloat the parameter lists, which might be less clean).